shopelectro/request_data.py

```diff
@@ -0,0 +1,71 @@
+import typing
+
+from django import http
+from django_user_agents.utils import get_user_agent
+
+
+class RequestData:
+    def __init__(
+        self, request: http.HttpRequest, url_kwargs: typing.Dict[str, str]
+    ):
+        self.request = request
+        self.url_kwargs = url_kwargs
+
+
+class ProductList(RequestData):
```

Comment:
```suggestion
class CatalogRequest(Request):
```

Reply:
i'll change it to `Catalog`. It's already in `request_data` module